### PR TITLE
chore(demo): prerender the pages of documentaion application

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [X] Build or CI related changes
- [ ] Documentation content changes

## What is the new behavior?
**SSG**

Read more:
* https://angular.io/guide/universal#why-use-server-side-rendering
* https://angular.io/guide/prerendering

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: projects/demo/routesFile.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/
+/getting-started/what-is-maskito
+/getting-started/maskito-libraries
+/core-concepts/overview
+/core-concepts/mask-expression
```
Comment (nsbarsukov) [inline projects/demo/routesFile.txt:5]: I will create a special utility function to generate this file from [DemoPath](https://github.com/Tinkoff/maskito/blob/main/projects/demo/src/app/demo-path.ts) later in separate PR.

File: projects/demo/project.json [hunk 2]
```diff
@@ -80,8 +80,9 @@
                     "echo 'Github pages require special baseHref + 404.html'",
                     "echo 'Read more: https://angular.io/guide/deployment#deploy-to-github-pages'",
                     "echo ------",
-                    "nx build --base-href='/maskito/'",
```
Comment (nsbarsukov) [inline projects/demo/project.json:83]: cli flag `--base-href='/maskito/'` does not properly work for prerender commands

File: projects/demo/project.json [hunk 3]
```diff
@@ -178,7 +179,7 @@
             "options": {
                 "browserTarget": "demo:build:production",
                 "serverTarget": "demo:server:production",
-                "routes": ["/"]
+                "routesFile": "./projects/demo/routesFile.txt"
```
Comment (nsbarsukov) [inline projects/demo/project.json:182]: * https://github.com/angular/universal/issues/1769

File: projects/demo/project.json [hunk 4]
```diff
@@ -68,6 +68,30 @@
...
+                        }
+                    ],
+                    "optimization": true,
+                    "outputHashing": "all",
+                    "sourceMap": false,
+                    "namedChunks": false,
+                    "aot": true,
+                    "extractLicenses": true,
+                    "vendorChunk": false,
+                    "buildOptimizer": true,
```
Comment (nsbarsukov) [inline projects/demo/project.json:86]: This line is copy-pasted from `production`-configuration.
I need to override only `baseHref` option and all other options from `production`.

Does anybody know a better way ?

Comment (splincode) [reply projects/demo/project.json:86]: `Does anybody know a better way ?`

`nx build -c github, production`

Comment (splincode) [reply projects/demo/project.json:86]: https://angular.io/guide/workspace-config
image

Comment (nsbarsukov) [reply projects/demo/project.json:86]: I will create issue to do it after upgrading of Nx.

Thanks!!

**Review by splincode — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/demo/routesFile.txt (hunk 1 above)
Comment (nsbarsukov) [reply projects/demo/routesFile.txt:5]: * https://github.com/Tinkoff/maskito/issues/240

File: projects/demo/project.json (hunk 4 above)
Comment (nsbarsukov) [reply projects/demo/project.json:86]: * https://github.com/Tinkoff/maskito/issues/241

**Pull request merged**